### PR TITLE
Developer guide for new structure data source integration

### PR DESCRIPTION
**Pull request by xzdandy**

### CODE REVIEW

**Comment by xzdandy** [comment]
@jiashenC Please review the documentation. Shall we ask for test cases when adding a new data source integration? If so, how do we do that given many of them needs to setup a backend server. Thanks!

**Comment by jiashenC** [comment]
> @jiashenC Please review the documentation. Shall we ask for test cases when adding a new data source integration? If so, how do we do that given many of them needs to setup a backend server. Thanks!

Yeah, for open sourced engine, we can probably setup server in CI as Postgres. For other commercial ones, we need to think of other solutions.

**Review by jiashenC — COMMENTED** [review]
Overall it looks good.

1. We probably need to explain where to add `requirements.txt` for installing native database-related libraries and how it is handled by us.
2. Just check the consistency of lower case / upper case for MyDB example.

File: docs/source/contribute/new-data-source.rst [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+- __init__.py
+- requirements.txt
+- mydb_handler.py
+
+The *__init__.py* can contain copyright information. The *requirements.txt* contains the extra python libraries that need to be installed via pip for the mydb data source. 
+
+.. note:: 
+
+   EvaDB will only install a data source's specific dependency libraries when a connection to the data source is created by the user via, e.g., `CREATE DATABASE mydb_source WITH ENGINE = "mydb";`.
```
Comment (xzdandy) [inline docs/source/contribute/new-data-source.rst:32]: @jiashenC This is what I have for requirements.txt. Could you help elaborate on that? Thanks!

**Comment by xzdandy** [comment]
Remove the unit test case requirements. We will add back when we have one. when we have one.

File: docs/source/contribute/new-data-source.rst (hunk 1 above)
Comment (jiashenC) [reply docs/source/contribute/new-data-source.rst:32]: Oh, i see. I missed that part.

**Review by jiashenC — APPROVED** [review]

**Pull request merged**